File: ImobDLApi/DTOs/ImovelComFiltros.cs
using ImobDLApi.models;

namespace ImobDLApi.DTOs
{
    public class ImovelComFiltros
    {
        public List<ImovelDTO> Imoveis { get; set; }
        public List<Bairro> Bairros { get; set; }
        public List<Tipo> Tipos { get; set; }
        public List<Cidade> Cidades { get; set; }
    }
}

File: client-app/src/app/DTOs/imovelComFiltros.ts
import { Bairro } from "../models/bairro";
import { Cidade } from "../models/cidade";
import { Imovel } from "../models/imovel";
import { Tipo } from "../models/tipo";

export default interface ImovelComFiltros{
    imoveis: Imovel[],
    bairros: Bairro[],
    cidades: Cidade[],
    tipos: Tipo[]
}